cmd/agent/daemon/app/app.go

```diff
@@ -383,6 +386,21 @@ func (a *App) Run(ctx context.Context) error {
 		}
 	}
 
+	// Load initial containers only after ebpf tracer startup. Tracer always listens
+	// to cgroup rmdir event. This reduces the chance of missing container removal events
+	// and leaving dangling cached containers in containers client.
+	// TODO: This sleep can be improved and instead check if at least one event is received from the kernel.
+	// But worst case if only cgroup events are tracked we still need some sleep.
+	sleep(ctx, 3*time.Second)
```

Comment:
Why exactly is this needed? 

The `containerRefreshLoop` is also loading all the containers. The first tick will come after `10s`(with default values). Shouldn't this be enough?

Reply:
The issue here that if we would load without any sleep there is a time window in which we may miss container delete event because cgroup rmdir is not yet attached. That means there is 10seconds window we may still contain cached container and ebpf event may get wrong context.